CMakeLists.txt

```diff
@@ -330,6 +330,42 @@ if ("${OS_NAME}" MATCHES "Linux")
       target_compile_definitions(ds2 PRIVATE HAVE_${CHECK})
     endif ()
   endforeach ()
+elseif ("${OS_NAME}" MATCHES "Windows")
+  set(CMAKE_EXTRA_INCLUDE_FILES windows.h)
+
+  include(CheckFunctionExists)
+  foreach (FUNC CreateRemoteThread GetVersionExA GetVersionExW
+      GetWindowsDirectoryA GetWindowsDirectoryW K32EnumProcesses
+      K32EnumProcessModules K32GetModuleBaseNameA K32GetModuleBaseNameW
+      K32GetModuleFileNameExA K32GetModuleFileNameExW GetModuleHandleA
+      GetModuleHandleW OpenProcessToken GetTokenInformation LookupAccountSidA
+      LookupAccountSidW CopySid GetLengthSid GetEnvironmentStringsW
+      FreeEnvironmentStringsW CreateProcessA CreateProcessW CreateThread
+      ExitThread ResumeThread TerminateThread SuspendThread GetThreadContext
+      SetThreadContext GetThreadPriority SetThreadPriority GetProcessId
+      GetThreadId OpenProcess OpenThread GetExitCodeProcess GetExitCodeThread
+      ReadProcessMemory WriteProcessMemory FlushInstructionCache
+      WaitForDebugEventEx ContinueDebugEvent TerminateProcess
+      DebugActiveProcess DebugActiveProcessStop VirtualAllocEx VirtualFreeEx
+      VirtualQueryEx CreateToolhelp32Snapshot Thread32First Thread32Next
+      SetUnhandledExceptionFilter)
+    CHECK_FUNCTION_EXISTS(${FUNC} HAVE_${FUNC})
```

Comment:
Is this going to automatically set the appropriate `WINAPI_FAMILY` macros, so that the definitions read from windows.h actually reflect the WinStore API restrictions?

Reply:
This won't set anything automatically, but I'm not sure I understand what
you're talking about. Do you have an example?

Reply:
What I meant was, the whole reason we need the header defining additional functions is because, when compiling for WinStore, certain prototypes and structs get hidden from Windows.h. The header makes the decision on whether to hide those or not based on the declared API family: WinStore apps are compiled with `-DWINAPI_FAMILY=WINAPI_FAMILY_APP`, and so for these checks to have the desired effect (i.e. determine whether certain prototypes are available when compiling for WinStore), the compilation for the checks also needs to pass those preprocessor flags.

I would imagine that cmake automatically adds those flags to every compile invocation when targeting WinStore, so it should just work, most probably. I just wanted to confirm that.

Reply:
Ok I see what you're talking about.

We never set these defines ourselves and as far as I can tell, they are set by the compiler based on a set of other flags (e.g.: which target is used, etc). Because of that, I think these checks should do the right thing.